### PR TITLE
Multi file image recognition

### PR DESCRIPTION
**Pull request by abhilashcherukat**
This program extent the functionality to extract images from a folder and map that to an name which is matched

### CODE REVIEW

**Comment by abhilashcherukat** [comment]
Please review my code and if fit add it to the examples

File: examples/multiple_facerec_from_webcam_faster.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+known_image=[]
+
+#ForEncoding
+known_face_encoding=[]
+
+#Files should be in the Imagefolder folder
+for file in os.listdir("Imagefolder"):
+    try:
+        #Extracting person name from the image filename eg:Abhilash.jpg
+        known_person.append(str(file).replace(".jpg", ""))
```
Comment (AlainPilon) [inline examples/multiple_facerec_from_webcam_faster.py]: isnt `file` always a string?

File: examples/multiple_facerec_from_webcam_faster.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    except Exception as e:
+        print "Not loaded"+str(e)
+        pass
+
+
+# Initialize some variables
+face_locations = []
+face_encodings = []
+face_names = []
+#process_this_frame = 0
```
Comment (AlainPilon) [inline examples/multiple_facerec_from_webcam_faster.py]: plz remove commented out testing code

File: examples/multiple_facerec_from_webcam_faster.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    if process_this_frame%5==0:
+        # Find all the faces and face encodings in the current frame of video
+        face_locations = face_recognition.face_locations(rgb_small_frame)
+        face_encodings = face_recognition.face_encodings(rgb_small_frame, face_locations)
+
+        face_names = []
+        Length=len(known_face_encoding)
+        for face_encoding in face_encodings:
+            match = face_recognition.compare_faces(known_face_encoding, face_encoding)
+            Matches=np.where(match)[0] #Checking which image is matched
```
Comment (AlainPilon) [inline examples/multiple_facerec_from_webcam_faster.py]: Dont capitalize variable name (typo?)

**Comment by christiansievers** [comment]
works like a charm, thanks. Except for line 38 which throws an error:

>   File "multiple_facerec_from_webcam_faster.py", line 38
>     print "Not loaded"+str(e)
>                      ^
> SyntaxError: invalid syntax

**Comment by vrushang25** [comment]
I am working in face recogniton with using the web camera and i am facing below problem.
My Face is recognized same as my brother's face. I tried to reduce the tolerance to the 0.55 but no effect. what should i do?

**Comment by zbot473** [comment]
(stupid question) 
Would this work with multiple people?

**Comment by abhilashcherukat** [comment]
Yes, You can take photo of the people you want to recognize and put it in
the knownface folder

Regards
*Abhilash Cherukat*
8147631545

**Comment by Hrishi1999** [comment]
Oh, I programmed the same thing, I hope I'd seen this earlier. Mine is a slight different.

`objlist = []
obj_img = []
obj_enc = []
obj_names = []
cnt = 0

for root, dirs, files in os.walk(os.getcwd()):
    for file in files:
        if file.endswith('.jpg'):
            result = ''.join([i for i in str(file.replace(".jpg","")) if not i.isdigit()])
            print(file)
            obj_names.append(result)
            objlist.append(file)
            try:
                obj_img.append(face_recognition.load_image_file(objlist[cnt]))
                obj_enc.append(face_recognition.face_encodings(obj_img[cnt])[0])
            except:
                print("Face not found in {0}".format(file))
                continue
            cnt = cnt +1

# Create arrays of known face encodings and their names
known_face_encodings = obj_enc
known_face_names = obj_names


# Initialize some variables
face_locations = []
face_encodings = []
face_names = []`

**Comment by ageitgey** [comment]
This is a good demo (aside from some small PEP8 code formatting issues), but I don't want to include lots of slightly different demos in the examples folder, especially if they are really demos of Python features (listing files in a directory) and not really different demos of the library itself. Otherwise I'm worried people will be confused over how they are supposed to use the library. The idea is that the examples I include should be super simple examples of how to use the API and then the user can re-think them as part of a bigger Python program.

So please feel free to post this demo yourself or post it as a comment on the Issues so anyone can find it with a search. But it's close enough to the existing demos that I don't to add it to the main examples folder. But thanks again for sharing!

**Pull request closed without merging**